tools/mksurfdata_esmf/README

```diff
@@ -0,0 +1,139 @@
+=======
+Purpose
+=======
+This tool is intended to generate fsurdat files (surface datasets) for the
+CTSM. It can generate global, regional, and single-point fsurdat files, as long
+as a mesh file is available for the grid.
+
+The subset_data tool allows users to make fsurdat files from existing fsurdat
+files when a mesh file is unavailable. Generally, users should consider the
+subset_data tool for generating regional and single-point fsurdat files.
+
+===================
+Build Requirements:
+===================
+
+mksurfdata_esmf is a distributed memory parallel program (using Message Passing
+Interface -- MPI) that utilizes both ESMF (Earth System Modelling Framework)
+for regridding as well as PIO (Parallel I/O) and NetCDF output. As
+such, libraries must be built for the following:
+
+1) MPI
+2) NetCDF
+3) PIO
+4) ESMF
+
+In addition for the build: python, bash-shell, CMake and GNU-Make are required
+
+These libraries need to be built such that they can all work together in the
+same executable. Hence, the above order may be required in building them.
+
+=========================================
+Use cime to manage the build requirements
+=========================================
+
+For users working on cime machines you can use the build script to build the
+tool. On other machines you'll need to do a port to cime and tell how to build
+for that machine. That's talked about in the cime documentation.
+And you'll have to make some modifications to the build script.
+
+https://github.com/ESMCI/cime/wiki/Porting-Overview
+
+Machines that already run CTSM or CESM have been ported to cime. So if you can
+run the model on your machine, you will be able to build the tool there.
+
```

Comment:
@slevis-lmwg this makes it sound like that mksurfdata_esmf can be run on any CESM supported machine. Which isn't really true now. In principle that is what we want, but in practice that doesn't work now.

I'll add a caveat for this in the README here.

Reply:
Resolved in #2417